src/main/java/edu/uci/ics/crawler4j/robotstxt/RobotstxtParser.java

```diff
@@ -59,7 +62,12 @@ public static HostDirectives parse(String content, String myUserAgent) {
 
       if (line.matches(PATTERNS_USERAGENT)) {
         String ua = line.substring(PATTERNS_USERAGENT_LENGTH).trim().toLowerCase();
-        inMatchingUserAgent = "*".equals(ua) || ua.contains(myUserAgent);
+        if (ua.equals("*") || ua.contains(myUserAgent)) {
+          inMatchingUserAgent = true;
+        } else {
+          inMatchingUserAgent = false;
+        }
```

Comment:
You are correct. However, I initially based these patches on commit dd9f014c35684343621b83902b8e4307d6a7eb34 where the old statement was in place. After rebasing, this apparently got reverted to the older version. I'll adjust my patch to fix this.